src/json-rpc/workspace-master-api.ts

```diff
@@ -249,6 +260,7 @@ export class WorkspaceMasterApi implements IWorkspaceMasterApi {
     private onConnectionClose(entryPoint: string): void {
         console.warn('WebSocket connection is closed.');
         if (this.reconnectionAttemptNumber === this.maxReconnectionAttempts) {
+            this.wsMasterEventEmitter.emit(this.webSocketFailedEventName, entryPoint);
```

Comment:
As far as I understand it allows Dashboard to display banner after 5 failure.
I would like if we improve the way how we reestablish websocket:
1. WebSocket is established.
2. WebSocket is interrupted.
 2.1 WsClient notifies the dashboard about it.
 2.2.Dashboard displays banner that WebSocket is lost
3. WsClient endless tries to reconnect with dynamic delay: first 10 times every second, then every 30 seconds for example.
4. Once Websocket is reestablished, WSClient notifies Dashboard that everything is fine with Websockets now.
5. Dashboard hides the banner.